docs/source/_src_user_guide/state_propagation/environment_setup/architecture/environment_during_propagation.rst

```diff
@@ -0,0 +1,131 @@
+.. _environment_during_propagation:
+
+===================================================
+Interacting with the environment during propagation
+===================================================
+
+Each body object and its constituent members is updated to the current state and time automatically during the numerical propagation. We stress that only those objects that are relevant for a given propagation are updated every time step. See :ref:`single_propagation_evaluation` for a more detailed discussion on what happens during a function evaluation of the state derivative model.
+
+Even though the environment is updated automatically, in various cases a user has control over *how* it gets updated. This is the case when using any of the :ref:`custom models <custom_models>` in Tudat. Typical examples of such models are aerodynamic guidance, or thrust guidance. When defining such custom models, you will in many cases need to access the properties of bodies in the environment to perform your calculations. This page lists how to access such properties, and how to use/interpret them (if relevant). Some examples are:
+
+* Retrieving the translational state of a spacecraft w.r.t. a central body to calculate its thrust direction 
+* Retrieving the current atmospheric density during entry to calculate the required vehicle orientation
+* Retrieving properties that cannot be saved as a dependent variable to calculate a custom dependent variable
+* ...
+
```

Comment:
here you mention three possible examples for which this documentation page would be useful. The first two are directly addressed by some subsections (how to retrieve translational state / current atmospheric density), while the third is more top-level. I understand why this is here, but it seems to me a bit decorrelated from the page content. Would it make sense to either add a small example showing how this could be done, or add a link to the documentation page where this is explicitely described?